jaxrs/src/main/java/com/webcohesion/enunciate/modules/jaxrs/JaxrsModule.java

```diff
@@ -16,241 +14,285 @@
+
+	private DataTypeDetectionStrategy defaultDataTypeDetectionStrategy;
+	private final List<MediaTypeDefinitionModule> mediaTypeModules = new ArrayList<MediaTypeDefinitionModule>();
+	private ApiRegistry apiRegistry;
+	static final String NAME = "jaxrs";
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	@Override
+	public List<DependencySpec> getDependencySpecifications() {
+		return Arrays.asList((DependencySpec) new MediaTypeDependencySpec());
+	}
+
+	public DataTypeDetectionStrategy getDataTypeDetectionStrategy() {
+		String dataTypeDetection = this.config.getString("[@datatype-detection]", null);
+
+		if (dataTypeDetection != null) {
+			try {
+				return DataTypeDetectionStrategy.valueOf(dataTypeDetection);
+			} catch (IllegalArgumentException e) {
+				// fall through...
+			}
+		}
+
+		if (this.defaultDataTypeDetectionStrategy != null) {
+			return this.defaultDataTypeDetectionStrategy;
+		}
+
+		if (this.enunciate.getIncludePatterns().isEmpty()) {
+			// if there are no configured include patterns, then we'll just
+			// stick with "local" detection so we don't include too much.
+			return DataTypeDetectionStrategy.local;
+		} else {
+			// otherwise, we'll assume the user knows what (s)he's doing and
+			// aggressively include everything.
+			return DataTypeDetectionStrategy.aggressive;
+		}
+	}
+
+	public void setDefaultDataTypeDetectionStrategy(DataTypeDetectionStrategy strategy) {
+		this.defaultDataTypeDetectionStrategy = strategy;
+	}
+
+	@Override
+	public void setApiRegistry(ApiRegistry registry) {
+		this.apiRegistry = registry;
+	}
+
+	public EnunciateJaxrsContext getJaxrsContext() {
+		return new EnunciateJaxrsContext(context);
+	}
+
+	@Override
+	public void call(EnunciateContext context) {
+
+		DataTypeDetectionStrategy detectionStrategy = getDataTypeDetectionStrategy();
+		Map<String, EnunciateJaxrsContext> contextMap = new HashMap<String, EnunciateJaxrsContext>();
+		if (detectionStrategy != DataTypeDetectionStrategy.passive) {
+			Set<? extends Element> elements = detectionStrategy == DataTypeDetectionStrategy.local
+					? context.getLocalApiElements() : context.getApiElements();
+			for (Element declaration : elements) {
+				EnunciateJaxrsContext jaxrsContext = null;
+				String relativeContextPath = null;
+				if (declaration instanceof TypeElement) {
+					TypeElement element = (TypeElement) declaration;
+
+					if ("org.glassfish.jersey.server.wadl.internal.WadlResource"
+							.equals(element.getQualifiedName().toString())) {
+						// known internal wadl resource not to be documented.
+						continue;
+					}
+
+					ApplicationPath applicationPathInfo = declaration.getAnnotation(ApplicationPath.class);
+					if (applicationPathInfo != null && !applicationPathInfo.value().trim().isEmpty()) {
+						relativeContextPath = applicationPathInfo.value();
+					}
+
+					ServiceContextRoot serviceCtx = declaration.getAnnotation(ServiceContextRoot.class);
```

Comment:
process the new annotation. i think its a problem if you dont want root context (like before)